src/workerd/io/tracer.c++

```diff
@@ -347,6 +347,11 @@ void WorkerTracer::setOutcome(EventOutcome outcome, kj::Duration cpuTime, kj::Du
   // All spans must have wrapped up before the outcome is reported. Report the top-level "worker"
   // span by deallocating its span builder.
   userRequestSpan = nullptr;
+
+  // Do not attempt to report an outcome event if logging is disabled, as with other event types.
+  if (pipelineLogLevel == PipelineLogLevel::NONE) {
```

Comment:
I'm not sure if PipelineLogLevel::NONE is widely used and if it is a mode that should be supported in the STW model... with this change we'd not be sending any events and effectively prevent the tail worker customEvent from being created, but we could also send only Onset and Outcome instead... I don't know how PipelineLogLevel::NONE came to be, so having more context would be helpful